### PR TITLE
[IA-4289]-add-tests-for-dataproc

### PR DESCRIPTION
**Pull request by ghost**
### Jira Ticket: https://broadworkbench.atlassian.net/browse/IA-4289

Connected to PROD-814

## Summary of changes:
Added 3 tests for dataproc in order to prevent a prod indecent from happening again.

### What
- Added 3 tests for dataproc, 2 creating dataproc spark instances, and one deleting a single node spark cluster
- Added a "getgoogledataproc" function to testData.ts to generate fake dataproc runtimes

### Why
- to prevent a prod indecent from happening again by getting proper coverage
- to test dataproc, which has historically been under-tested

### Testing strategy
- [ ] Create dataproc spark cluster and check if it has the proper values
- [ ] Create dataproc spark single node cluster and check if it has the proper values
- [ ] Remove dataproc spark single node cluster with no disk

### CODE REVIEW

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js [hunk 1]
```diff
@@ -589,6 +590,168 @@ describe('ComputeModal', () => {
   //   }))
   // })
 
+  // should create dataproc spark cluster successfully
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js]: Nit: Comment

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js [hunk 2]
```diff
@@ -589,6 +590,168 @@ describe('ComputeModal', () => {
...
+          numberOfPreemptibleWorkers: defaultNumDataprocPreemptibleWorkers,
+          cloudService: 'DATAPROC',
+          region: 'us-central1',
+          componentGatewayEnabled: true,
+        }),
+      })
+    );
+  });
+
+  // should create dataproc spark single node successfully
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js]: Nit: Comment 2

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js [hunk 3]
```diff
@@ -589,6 +590,168 @@ describe('ComputeModal', () => {
...
+          masterDiskSize: defaultDataprocMasterDiskSize,
+          cloudService: 'DATAPROC',
+          region: 'us-central1',
+          componentGatewayEnabled: true,
+        }),
+      })
+    );
+  });
+
+  // should delete spark single node successfully
```
Comment (n8landolt) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js]: Nit: Comment 3

**Review by n8landolt — APPROVED** [review]
Clean up those three comments and you're good to go!

**Review by lucymcnatt — APPROVED** [review]

**Pull request merged**